checker/src/org/checkerframework/checker/index/indexof/IndexOfAnnotatedTypeFactory.java

```diff
@@ -0,0 +1,130 @@
+    private final class IndexOfQualifierHierarchy extends MultiGraphQualifierHierarchy {
+
+        public IndexOfQualifierHierarchy(MultiGraphQualifierHierarchy.MultiGraphFactory factory) {
+            super(factory);
+        }
+
+        @Override
+        public AnnotationMirror greatestLowerBound(AnnotationMirror a1, AnnotationMirror a2) {
+            if (AnnotationUtils.areSame(a1, UNKNOWN)) {
+                return a2;
+            }
+            if (AnnotationUtils.areSame(a2, UNKNOWN)) {
+                return a1;
+            }
+            if (AnnotationUtils.areSame(a1, BOTTOM)) {
+                return a1;
+            }
+            if (AnnotationUtils.areSame(a2, BOTTOM)) {
+                return a2;
+            }
+            UBQualifier a1Obj = UBQualifier.createUBQualifier(a1);
+            UBQualifier a2Obj = UBQualifier.createUBQualifier(a2);
+            UBQualifier glb = a1Obj.glb(a2Obj);
+            return convertUBQualifierToAnnotation(glb);
+        }
+
+        @Override
+        public AnnotationMirror leastUpperBound(AnnotationMirror a1, AnnotationMirror a2) {
+            if (AnnotationUtils.areSame(a1, UNKNOWN)) {
+                return a1;
+            }
+            if (AnnotationUtils.areSame(a2, UNKNOWN)) {
+                return a2;
+            }
+            if (AnnotationUtils.areSame(a1, BOTTOM)) {
+                return a2;
+            }
+            if (AnnotationUtils.areSame(a2, BOTTOM)) {
+                return a1;
+            }
+            UBQualifier a1Obj = UBQualifier.createUBQualifier(a1);
+            UBQualifier a2Obj = UBQualifier.createUBQualifier(a2);
+            UBQualifier lub = a1Obj.lub(a2Obj);
+            return convertUBQualifierToAnnotation(lub);
+        }
+
+        @Override
+        public boolean isSubtype(AnnotationMirror subAnno, AnnotationMirror superAnno) {
+            if (AnnotationUtils.areSameByClass(superAnno, IndexOfUnknown.class)) {
+                return true;
+            }
+            if (AnnotationUtils.areSameByClass(subAnno, IndexOfBottom.class)) {
+                return true;
+            }
+            if (AnnotationUtils.areSameByClass(subAnno, IndexOfUnknown.class)) {
+                return false;
+            }
+            if (AnnotationUtils.areSameByClass(superAnno, IndexOfBottom.class)) {
+                return false;
+            }
+
+            UBQualifier subtype = UBQualifier.createUBQualifier(subAnno);
+            UBQualifier supertype = UBQualifier.createUBQualifier(superAnno);
+            return subtype.isSubtype(supertype);
+        }
+    }
+
+    public AnnotationMirror convertUBQualifierToAnnotation(UBQualifier qualifier) {
+        if (qualifier.isUnknown()) {
+            return UNKNOWN;
+        } else if (qualifier.isBottom()) {
+            return BOTTOM;
+        }
+
+        LessThanLengthOf ltlQualifier = (LessThanLengthOf) qualifier;
+        return ltlQualifier.convertToIndexOfAnnotationMirror(processingEnv);
```

Comment:
The call is to a method whose name ends `to...AnnotationMirror`, but it's within a method whose name ends `toAnnotation`.  Please make the names consistent throughout the codebase.